src/App.js

```diff
@@ -84,9 +112,11 @@ export default class App extends Component {
           <Router>
             <AppHeader profile={this.state.profile} />
             <Switch>
-              <PrivateRoute exact path="/" component={Leaderboard} isAuthenticated={this.state.isAuthenticated} />
-              <Route path="/profile" >
-                <Profile isAuthenticated={this.state.isAuthenticated} user={this.state.user} profile={this.state.profile} authenticate={this.authenticate} signout={this.signout} />
+              <PrivateRoute exact path="/:groupId" isAuthenticated={this.state.isAuthenticated}>
+                <Leaderboard groups={this.state.groups} jwt={this.state.jwt} />
+              </PrivateRoute>
+              <Route path="/" >
+                <Profile isAuthenticated={this.state.isAuthenticated} user={this.state.user} groups={this.state.groups} profile={this.state.profile} authenticate={this.authenticate} updateGroups={this.updateGroups} signout={this.signout} />
```

Comment:
we might want to think about redux at some point. might be nice to store user, groups, profile etc globally.